File: python/problem1487.py
class Solution:
    def getFolderNames(self, names: List[str]) -> List[str]:
        name2k = {}
        result = []
        for name in names:
            if name not in name2k:
                result.append(name)
                name2k[name] = 0
            else:
                k = name2k[name] + 1
                s = name + '(' + str(k) + ')'
                while s in name2k:
                    k += 1
                    s = name + '(' + str(k) + ')'
                result.append(s)
                name2k[s] = 0
                name2k[name] = k
        return result

File: Java/problem1487.java
class Solution {
    public String[] getFolderNames(String[] names) {
        String[] result = new String[names.length];
        Map<String, Integer> map = new HashMap<>();
        for (int i = 0; i < names.length; ++i) {
            if (!map.containsKey(names[i])) {
                result[i] = names[i];
                map.put(names[i], 0);
            }
            else {
                int k = map.get(names[i]) + 1;
                String s = names[i] + "(" + k + ")";
                while (map.containsKey(s)) {
                    s = names[i] + "(" + (++k) + ")";
                }
                result[i] = s;
                map.put(names[i], k);
                map.put(s, 0);
            }
        }
        return result;
    }
}
